expression/builtin_cast.go

```diff
@@ -1348,7 +1348,8 @@ func WrapWithCastAsDecimal(expr Expression, ctx context.Context) Expression {
 // of expr is not type string,
 // otherwise, returns `expr` directly.
 func WrapWithCastAsString(expr Expression, ctx context.Context) Expression {
-	if expr.GetTypeClass() == types.ClassString {
+	exprTp := expr.GetType().Tp
+	if expr.GetTypeClass() == types.ClassString && !types.IsTypeTime(exprTp) && exprTp != mysql.TypeDuration {
```

Comment:
Is this related with greatest/least?

Reply:
It is better to add a comment here.

Reply:
It's a bug fix,
if we invoke evalString on a TimeExpr, it would raise an error.